### PR TITLE
Do not initialize model if it exists in CNN

### PR DESCRIPTION
**Pull request by nsorros**
Description
---

The rational here is to allow to build a model that powers `CNNClassifier` without a call to fit. The use case is a particular experiment I want to run which performs training on a subset of labels and then keeps all layers but the last and continues training with more labels as a warm up strategy. To do that, I need a way to build the model with more outputs before copying the weights from the trained model.

Maybe a better way would be to pass the weight layers in `fit` and use them for initialisation. What do you think @aCampello @ivyleavedtoadflax ? In any case, this definitely allows me to run the experiment for now but keen to hear what you think would be the best way to express this

Checklist
---

- [ ] Added link to Github issue or Trello card
- [x] Added tests

### CODE REVIEW

**Review by aCampello — APPROVED** [review]
Looks reasonable for me to store `self.model` so you can re-use and re-build.

File: wellcomeml/ml/cnn.py [hunk 1]
```diff
@@ -297,11 +297,15 @@ def fit(self, X, Y=None, embedding_matrix=None, steps_per_epoch=None):
...
         val_data = data.skip(steps_per_epoch)
 
-        strategy = self._get_distributed_strategy()
-        with strategy.scope():
-            self.model = self._build_model(
-                self.sequence_length, self.vocab_size, self.nb_outputs,
-                steps_per_epoch, embedding_matrix)
+        if hasattr(self, "model"):
+            logger.warning("Using existing model")
+            print("Using existing model")
```
[the remark this answers is not in the document]
Comment (nsorros) [reply wellcomeml/ml/cnn.py:302]: oops, this needs to be deleted indeed

**Comment by nsorros** [comment]
> Looks reasonable for me to store `self.model` so you can re-use and re-build.

Note that I was doing that before but I was not exposing the function `build_model` and additionally I was rebuilding in every call of fit whereas now if a model exists I do not rebuild to allow for additional training or pretrain of some sort

File: wellcomeml/ml/cnn.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (nsorros) [reply wellcomeml/ml/cnn.py]: b8defc4

**Pull request merged**